be/src/exec/vectorized/sorting/sort_column.cpp

```diff
@@ -453,7 +453,14 @@ Status sort_vertical_chunks(const std::atomic<bool>& cancel, const std::vector<C
     }
 
     if (limit < perm.size()) {
```

Comment:
This limit is a postpone prune, maybe you could pushdown the limit procedure into the `sort_vertical_columns` to limit earlier?

Reply:
That could save some work if many columns exist

Reply:
Thanks, good suggestion. I'll update

Reply:
I found that there is no need to pushdown the procedure, cause VerticalColumnSorter has already taken equal ranges into account